go/def.bzl

```diff
@@ -382,16 +287,26 @@ def go_library_impl(ctx):
     _emit_go_asm_action(ctx, src, asm_hdrs, obj)
     extra_objects += [obj]
 
-  out_lib = ctx.outputs.lib
+  lib_name = _go_importpath(ctx) + ".a"
+  out_lib = ctx.new_file(lib_name)
+  search_path = out_lib.path[:-len(lib_name)]
   gc_goopts = _gc_goopts(ctx)
-  _emit_go_compile_action(ctx, go_srcs, deps, out_lib, extra_objects, gc_goopts)
-
-  transitive_libs = set([out_lib])
-  transitive_importmap = {out_lib.path: _go_importpath(ctx)}
+  transitive_libs = depset([out_lib])
```

Comment:
Might be good to name these `transitive_go_libraries` and `transitive_go_library_paths` to be consistent with the provider.